### PR TITLE
[gestalt][gestalt-datepicker] 55.2.0 update

### PR DESCRIPTION
**Pull request by zifahm**
Changes reflect from 53.1 to 55.2.0
https://gestalt.netlify.app/whats_new

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## gestalt-datepicker ([<kbd>unpkg</kbd>](https://unpkg.com/browse/gestalt-datepicker@latest/))
was missing the following properties:
1. Fragment
1. jsx
1. jsxs
## gestalt ([<kbd>unpkg</kbd>](https://unpkg.com/browse/gestalt@latest/))
was missing the following properties:
1. Fragment
1. jsx
1. jsxs
1. Collection
1. DeviceTypeProvider

<details>
<summary>as well as these 6 other properties...</summary>
<p>ExperimentProvider, I18nProvider, MasonryDefaultLayout, MasonryUniformRowLayout, ScrollFetch, useColorScheme</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5ae1c83fd20634c9b32f4758ea9768397615ea0a
</p>

File: types/gestalt/index.d.ts [hunk 1]
```diff
@@ -415,24 +415,28 @@ export interface ComboBoxProps {
...
-    onFocus?: (args: { event: React.FocusEvent<HTMLInputElement>; value: string }) => void;
-    onKeyDown?: (args: { event: React.KeyboardEvent<HTMLInputElement>; value: string }) => void;
-    onClear?: () => void;
+    zIndex?: Indexable | undefined;
+    disabled?: boolean | undefined;
+    errorMessage?: string | undefined;
+    helperText?: string | undefined;
+    inputValue?: string | undefined;
+    labelDisplay?: 'visible' | 'hidden' | undefined;
+    onChange?: (args: { event: React.SyntheticEvent<HTMLInputElement>; value: string }) => void | undefined;
```
Comment (gabritto) [inline types/gestalt/index.d.ts:424]: did you mean for the type of those methods e.g. `onChange` to be `((args: { event: React.SyntheticEvent<HTMLInputElement>; value: string }) => void) | undefined`? the way they are written means they can *return* `undefined`, not that the methods themselves can *be* `undefined`.

**Review by gabritto — CHANGES_REQUESTED** [review]
looks good except for the comment I left

File: types/gestalt/index.d.ts (hunk 1 above)
Comment (zifahm) [reply types/gestalt/index.d.ts:424]: Yeah, I'm missig brackets [😅](https://emojipedia.org/grinning-face-with-sweat/)

**Review by gabritto — APPROVED** [review]

**Comment by zifahm** [comment]
Ready to merge

**Pull request merged**